.github/workflows/publish-lutris-ppa.yml

```diff
@@ -48,10 +99,14 @@ jobs:
         env:
           PPA_GPG_PRIVATE_KEY: ${{ secrets.PPA_GPG_PRIVATE_KEY }}
           PPA_GPG_PASSPHRASE: ${{ secrets.PPA_GPG_PASSPHRASE }}
+          # This will be passed through to the build-specific script for
+          # Jammy and causes it to produce additional builds for the
+          # Ubuntu codenames listed below.
+          SERIES: "kinetic lunar"
         run: |
```

Comment:
I see you are making the staging-ppa match release-ppa. Okay.

But this is a maintenance burden; any change must be reflected in both sections, yes?

Better if these bits could be shared. If not, a comment describing what needs to be match would at least be *something*. It's not clear to me what must match between the two sections, and what must differ.

Reply:
Yeeeeah... the reason I had to split them was because the versions could be different, and joining them together was causing problems.. I was lazy and really need to comment about it though.. I'll do that with my next commit.

Reply:
I'm trying to rethink this.. and I think a way that I could combine these into a single section would be if I store my builds under separate directories for each PPA I'm producing builds for and then just push out to any builds under those directories...

So I could set environment variables like:
```yml
    env:
      PPA_URI: "ppa:lutris-team/lutris"
      PPA_STAGING_URI: "ppa:lutris-team/lutris-staging"
      IS_PRERELEASE: ${{ github.event.release.prerelease }}
```

And then, in the build-ubuntu.sh script I could run a the first time against PPA_URI and store the builds under `../releases/` and if  IS_PRERELEASE is true, I could run a second time against the PPA_STAGING_URI and store the builds under `../prereleases/` and then `dput` everything under `../releases/` to PPA_URI and everything under `../prereleases/` to PPA_STAGING_URI.

I think that would work.. what do you think?

Reply:
The pros and cons of the alternate approach would be..

**Pros**
- cleaner github workflow, and one shared section.
- easier maintenance?

**Cons**
- adds a bit more complication to the build script.
- the build process would take a bit longer under a shared runner (building both release and prerelease on the same runner) than they would have processing in parallel as separate jobs on separate runners.

Reply:
Actually.. I think I can use this to make it cleaner:

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defaults

Reply:
That sounds like a good idea. I do not think concurrency is an issue- surely Lutris is cheap to build in any case and we don't need to throw cores at it.

Reply:
I can make these changes to the `publish-lutris-ppa.yml` without any changes to the build scripts:
```yml
# Inspired by
# https://github.com/yuezk/GlobalProtect-openconnect/blob/master/.github/workflows/build.yml
name: Publish Lutris PPA

on:
  # See below for details about the 'release' event.
  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release
  release:
    types: [released,prereleased]

# These options are shared between all jobs.
defaults:
  strategy:
    # GitHub only has runners for LTS versions of Ubuntu.
    # https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
    matrix:
      os: [ ubuntu-22.04, ubuntu-20.04 ]
  runs-on: ${{ matrix.os }}
  env:
    PPA_RELEASE_URI: "ppa:lutris-team/lutris"
    PPA_PRERELEASE_URI: "ppa:lutris-team/lutris-staging"
    # This will be passed through to the build-specific script for
    # Jammy and causes it to produce additional builds for the
    # Ubuntu codenames listed below.
    SERIES: "kinetic lunar"

jobs:
  # Only build the release PPA when the event is not a prerelease.
  release-ppa:
    if: github.event.release.prerelease == false
    steps:
      - name: Checkout the repository
        uses: actions/checkout@v2

      - name: Install build dependencies
        run: |
          ./.github/scripts/install-ubuntu.sh

      # This builds and signs the debian package for each OS in matrix.os
      # This script also recalculates dependencies based on the current
      # OS and can potentially produce different package control files,
      # but these files are not kept.
      #
      # Signing packages uses the secrets referenced below and passes
      # them through the build-ubuntu.sh script and into the "github-ppa"
      # Makefile directive.
      #
      # It would probably be a good idea to have a unique GPG key just
      # for this process, but whatever GPG key is passed here _must_ be
      # registered on the PPA as an authorized key.
      - name: Build the Debian Release Package
        env:
          PPA_URI: ${{ env.PPA_RELEASE_URI }}
          PPA_GPG_PRIVATE_KEY: ${{ secrets.PPA_GPG_PRIVATE_KEY }}
          PPA_GPG_PASSPHRASE: ${{ secrets.PPA_GPG_PASSPHRASE }}
        run: |
          ./.github/scripts/build-ubuntu.sh

      # Pushes the build to the Release PPA.
      - name: Publish to Release PPA
        run: |
          dput ${{ env.PPA_RELEASE_URI }} ../*.changes

  # This is the same as the release-ppa job, but the build process
  # environment variable PPA_URI targets PPA_PRERELEASE_URI instead.
  staging-ppa:
    steps:
      - name: Checkout the repository
        uses: actions/checkout@v2

      - name: Install build dependencies
        run: |
          ./.github/scripts/install-ubuntu.sh

      # Override PPA_URI with PPA_STAGING_URI so that our versions
      # are correctly targeted for that PPA.
      # PPA_GPG_PRIVATE_KEY & PPA_GPG_PASSPHRASE are required for
      # signing the packages.
      - name: Build Debian Prerelease Package
        env:
          PPA_URI: ${{ env.PPA_PRERELEASE_URI }}
          PPA_GPG_PRIVATE_KEY: ${{ secrets.PPA_GPG_PRIVATE_KEY }}
          PPA_GPG_PASSPHRASE: ${{ secrets.PPA_GPG_PASSPHRASE }}
        run: |
          ./.github/scripts/build-ubuntu.sh

      # Pushes the build to the Staging PPA.
      - name: Publish to Staging PPA
        run: |
          dput ${{ env.PPA_PRERELEASE_URI }} ../*.changes
```

Or,  I can combine them into a single job.
```yml
# Inspired by
# https://github.com/yuezk/GlobalProtect-openconnect/blob/master/.github/workflows/build.yml
name: Publish Lutris PPA

on:
  # See below for details about the 'release' event.
  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release
  release:
    types: [released,prereleased]

# These options are shared between all jobs.
defaults:
  strategy:
    # GitHub only has runners for LTS versions of Ubuntu.
    # https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
    matrix:
      os: [ ubuntu-22.04, ubuntu-20.04 ]
  runs-on: ${{ matrix.os }}
  env:
    PPA_RELEASE_URI: "ppa:lutris-team/lutris"
    PPA_PRERELEASE_URI: "ppa:lutris-team/lutris-staging"
    # This will be passed through to the build-specific script for
    # Jammy and causes it to produce additional builds for the
    # Ubuntu codenames listed below.
    SERIES: "kinetic lunar"

jobs:
  # Only build the release PPA when the event is not a prerelease.
  publish-ppa:
    steps:
      - name: Checkout the repository
        uses: actions/checkout@v2

      - name: Install build dependencies
        run: |
          ./.github/scripts/install-ubuntu.sh

      # This builds and signs the debian package for each OS in matrix.os
      # This script also recalculates dependencies based on the current
      # OS and can potentially produce different package control files,
      # but these files are not kept.
      #
      # Signing packages uses the secrets referenced below and passes
      # them through the build-ubuntu.sh script and into the "github-ppa"
      # Makefile directive.
      #
      # It would probably be a good idea to have a unique GPG key just
      # for this process, but whatever GPG key is passed here _must_ be
      # registered on the PPA as an authorized key.
      - name: Build the Debian Release Package
        if: github.event.release.prerelease == false
        env:
          PPA_URI: ${{ env.PPA_RELEASE_URI }}
          PPA_GPG_PRIVATE_KEY: ${{ secrets.PPA_GPG_PRIVATE_KEY }}
          PPA_GPG_PASSPHRASE: ${{ secrets.PPA_GPG_PASSPHRASE }}
        run: |
          ./.github/scripts/build-ubuntu.sh

      # Pushes the build to the Release PPA.
      - name: Publish to Release PPA
        if: github.event.release.prerelease == false
        run: |
          dput ${{ env.PPA_RELEASE_URI }} ../*.changes

      # Override PPA_URI with PPA_STAGING_URI so that our versions
      # are correctly targeted for that PPA.
      # PPA_GPG_PRIVATE_KEY & PPA_GPG_PASSPHRASE are required for
      # signing the packages.
      - name: Build Debian Prerelease Package
        env:
          PPA_URI: ${{ env.PPA_PRERELEASE_URI }}
          PPA_GPG_PRIVATE_KEY: ${{ secrets.PPA_GPG_PRIVATE_KEY }}
          PPA_GPG_PASSPHRASE: ${{ secrets.PPA_GPG_PASSPHRASE }}
        run: |
          ./.github/scripts/build-ubuntu.sh

      # Pushes the build to the Staging PPA.
      - name: Publish to Staging PPA
        run: |
          dput ${{ env.PPA_PRERELEASE_URI }} ../*.changes
```

Reply:
I went with the first example from above in https://github.com/lutris/lutris/pull/4653/commits/d06a5422a70291b57d98811d74c952357c86857e, but please let me know if the latter is more preferable, and I'll change it to that instead.